pkg/apis/serving/v1/route_lifecycle.go

```diff
@@ -174,10 +174,13 @@ func (rs *RouteStatus) MarkCertificateReady(name string) {
 
 // MarkCertificateNotReady marks the RouteConditionCertificateProvisioned
 // condition to indicate that the Certificate is not ready.
-func (rs *RouteStatus) MarkCertificateNotReady(name string) {
+func (rs *RouteStatus) MarkCertificateNotReady(c *v1alpha1.Certificate) {
+	certificateCondition := c.Status.GetCondition("Ready")
+	certMessage := certificateCondition.GetReason()
+
 	routeCondSet.Manage(rs).MarkUnknown(RouteConditionCertificateProvisioned,
 		"CertificateNotReady",
-		"Certificate %s is not ready.", name)
+		"Certificate %s is not ready. Certificate Message: %s", c.Name, certMessage)
```

Comment:
This would apply to `MarkCertificateProvisionFailed`  - not here

```suggestion
		"Certificate %s is not ready: %s", c.Name, certMessage)
```